File: src/JavaScript/easy/string/problem1446.js
/**
 * @param {string} s
 * @return {number}
 */
var maxPower = function (s) {
    let counter = 1
    let max = 1;
    for (let i = 0; i < s.length; i++) {
      if (i + 1 <= s.length) {
        if (s[i] == s[i + 1]) {
          counter++
        } else {
          if(max < counter){
              max = counter
          }
          counter = 1
        }
      }
    }
    return max
  };
  
  // let s = "leetcode";
  // s = "abbcccddddeeeeedcba"
  // let s = "ccbccbb"
  let s = "cbcb"
  
  maxPower(s);
  

File: src/Python/easy/string/problem1446.py
class Solution:
    def maxPower(self, s: str) -> int:
        max_num = 1
        counter = 1
        for i in range(len(s) - 1):
            if s[i] == s[i + 1]:
                counter += 1
                if max_num < counter:
                    max_num = counter
            else:
                counter = 1
        return print(max_num)



s = "abbcccddddeeeeedcba"
s = "cc"

solution = Solution()
solution.maxPower(s)